### ISSUE TITLE
使用create_parameter 创建的参数在静态图中不能被训练

### ISSUE THREAD

**Issue by zzdang** [PaddlePaddle/Paddle#27691]
- 版本、环境信息：
    1）PaddlePaddle版本：1.7
    2）GPU：10.2
    4）系统环境：python2.7 centos

- 训练信息
    1）单机，单卡
    2）Operator信息 layers.create_parameter
- 问题描述：请详细描述您的问题，同步贴出报错信息、日志、可复现的代码片段:
使用create_parameter创建了一个参数，给pooling层额外加了一个平均的参数，在训练中找不到该参数使用main_prog.global_block().iter_parameters()
网络结构如下：
class EfficientNet():
    def __init__(self, name='b0', padding_type='SAME', override_params=None, is_test=False, use_se=True):
        valid_names = ['b' + str(i) for i in range(8)]
        assert name in valid_names, 'efficient name should be in b0~b7'
        model_name = 'efficientnet-' + name
        self._blocks_args, self._global_params = get_model_params(model_name, override_params)
        self._bn_mom = self._global_params.batch_norm_momentum
        self._bn_eps = self._global_params.batch_norm_epsilon
        self.is_test = is_test
        self.padding_type = padding_type
        self.use_se = use_se
        ### GEM
        self.pool_type = "gem"
        #with fluid.dygraph.guard():
        self.gem_param = layers.create_parameter(shape=[1], dtype='float32', name='gem', \
               default_initializer=fluid.initializer.ConstantInitializer(value=3.0))
        self.eps = 1e-6
        self.gem = gem()

    def net(self, input, embedding_size=1000, is_test=False):

        conv = self.extract_features(input, is_test=is_test)

        out_channels = round_filters(1280, self._global_params)
        conv = self.conv_bn_layer(conv,
                                  num_filters=out_channels,
                                  filter_size=1,
                                  bn_act='swish',
                                  bn_mom=self._bn_mom,
                                  bn_eps=self._bn_eps,
                                  padding_type=self.padding_type,
                                  name='',
                                  conv_name='_conv_head',
                                  bn_name='_bn1')

        if not self.pool_type == "gem":
            pool = fluid.layers.pool2d(input=conv, pool_type='avg', global_pooling=True, use_cudnn=False)

            if self._global_params.dropout_rate:
                pool = fluid.layers.dropout(pool, self._global_params.dropout_rate, dropout_implementation='upscale_in_train')
        else:
            fluid.layers.Print(self.gem_param, message="~~~~~~" )#, print_phase='backward')
            pool = layers.pow( fluid.layers.pool2d(input=layers.pow( \
                    layers.clip(conv, min=self.eps, max=sys.float_info.max), factor=self.gem_param), pool_type='avg', \
                global_pooling=True, use_cudnn=False), factor = 1./self.gem_param)

        if embedding_size > 0:
            embedding = fluid.layers.fc(input=pool, size=embedding_size)
            return embedding
        else:
            return pool

**Comment by jerrywgz** [comment]
可以麻烦提供一个可以复现的最小case吗？

**Comment by zzdang** [comment]
可能有点麻烦，现在不确定在静态图里面create_parameter该放在哪里，with fluid.program_guard(main_prog, startup_prog) 这个后面还是 with fluid.unique_name.guard()？您的Hi账号可以加一下嘛？

**Comment by zzdang** [comment]
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import os
import sys,time
import numpy as np
import paddle
import paddle.fluid as fluid


def optimizer_setting(params):

    bd = [int(e) for e in params["lr_steps"].split(',')]
    base_lr = params["lr"]
    lr = [base_lr * (0.5 ** i) for i in range(len(bd) + 1)]
    optimizer = fluid.optimizer.Momentum(
        learning_rate=fluid.layers.piecewise_decay(
            boundaries=bd, values=lr),
        momentum=0.9,
        regularization=fluid.regularizer.L2Decay(1e-4))
    return optimizer



startup_prog = fluid.Program()
train_prog = fluid.Program()

with fluid.program_guard(train_prog, startup_prog):

    with fluid.unique_name.guard():
        gem_param = fluid.layers.create_parameter(shape=[1], dtype='float32', name='gem', \
               default_initializer=fluid.initializer.ConstantInitializer(value=3.0))        
        avg_cost = 2 * gem_param
        params = {}
        params["lr"] = 0.1
        params["lr_steps"] = "1000,10000"
        params["name"] = "piecewise_decay"
        optimizer = optimizer_setting(params)
        optimizer.minimize(avg_cost)
        global_lr = optimizer._global_learning_rate()

pd_weights = ['gem']

for param in train_prog.global_block().all_parameters():
    if param.name not in pd_weights:
        print("load: {}".format(param.name))
        param.stop_gradient = True
    else:
        print("param train", param)

train_fetch_list = [global_lr.name, avg_cost.name]

place = fluid.CUDAPlace(0)
exe = fluid.Executor(place)

exe.run(startup_prog)

train_exe = fluid.ParallelExecutor(
    main_program=train_prog,
    use_cuda=True,
    loss_name=avg_cost.name)

totalruntime = 0
iter_no = 0
train_info = [0, 0, 0, 0]
while iter_no <= 100:
    t1 = time.time()
    lr, loss = train_exe.run(fetch_list=train_fetch_list)
    print(iter_no, lr, loss)
    iter_no += 1

**Comment by zzdang** [comment]
`from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import os
import sys,time
import numpy as np
import paddle
import paddle.fluid as fluid


def optimizer_setting(params):

    bd = [int(e) for e in params["lr_steps"].split(',')]
    base_lr = params["lr"]
    lr = [base_lr * (0.5 ** i) for i in range(len(bd) + 1)]
    optimizer = fluid.optimizer.Momentum(
        learning_rate=fluid.layers.piecewise_decay(
            boundaries=bd, values=lr),
        momentum=0.9,
        regularization=fluid.regularizer.L2Decay(1e-4))
    return optimizer



startup_prog = fluid.Program()
train_prog = fluid.Program()

with fluid.program_guard(train_prog, startup_prog):

    with fluid.unique_name.guard():
        gem_param = fluid.layers.create_parameter(shape=[1], dtype='float32', name='gem', \
               default_initializer=fluid.initializer.ConstantInitializer(value=3.0))        
        avg_cost = 2 * gem_param
        params = {}
        params["lr"] = 0.1
        params["lr_steps"] = "1000,10000"
        params["name"] = "piecewise_decay"
        optimizer = optimizer_setting(params)
        optimizer.minimize(avg_cost)
        global_lr = optimizer._global_learning_rate()

pd_weights = ['gem']

for param in train_prog.global_block().all_parameters():
    if param.name not in pd_weights:
        print("load: {}".format(param.name))
        param.stop_gradient = True
    else:
        print("param train", param)

train_fetch_list = [global_lr.name, avg_cost.name]

place = fluid.CUDAPlace(0)
exe = fluid.Executor(place)

exe.run(startup_prog)

train_exe = fluid.ParallelExecutor(
    main_program=train_prog,
    use_cuda=True,
    loss_name=avg_cost.name)

totalruntime = 0
iter_no = 0
train_info = [0, 0, 0, 0]
while iter_no <= 100:
    t1 = time.time()
    lr, loss = train_exe.run(fetch_list=train_fetch_list)
    print(iter_no, lr, loss)
    iter_no += 1



`

**Comment by zzdang** [comment]
![image](https://user-images.githubusercontent.com/22818954/94525253-6509e880-0266-11eb-9efa-fd91f5f5c5ab.png)
这个是打印信息

**Comment by jerrywgz** [comment]
根据复现代码在1.8.5和2.0beta是正常的
![image](https://user-images.githubusercontent.com/22989727/94536523-4b23d200-0275-11eb-890c-a17fc954a12f.png)

**Comment by zzdang** [comment]
好的，谢谢
